File: CSharp/SpecStudioExamplesTestsCSharp/common/ReplacePostTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class ReplacePostTyped
    {
        public int id;
        public int userId;
        public string title;
        public string body;

        public ReplacePostTyped(int id, int userId, string title, string body)
        {
            this.id = id;
            this.userId = userId;
            this.title = title;
            this.body = body;
        }

        public ReplacePostString ToReplacePostString()
        {
            return new ReplacePostString(
                Json.ToText(this.id),
                Json.ToText(this.userId),
                Json.ToText(this.title),
                Json.ToText(this.body)
            );
        }

        public static List<ReplacePostString> ToStringList(List<ReplacePostTyped> list)
        {
            var result = new List<ReplacePostString>();
            foreach (var t in list) result.Add(t.ToReplacePostString());
            return result;
        }

        public static List<ReplacePostTyped> FromStringList(List<ReplacePostString> list)
        {
            var result = new List<ReplacePostTyped>();
            foreach (var s in list) result.Add(s.ToReplacePostTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteNumber("id", this.id);
            w.WriteNumber("userId", this.userId);
            w.WriteString("title", this.title);
            w.WriteString("body", this.body);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static ReplacePostTyped FromJsonElement(JsonElement m)
        {
            return new ReplacePostTyped(
                Json.AsInt(Json.Require(m, "id"), "id"),
                Json.AsInt(Json.Require(m, "userId"), "userId"),
                Json.AsString(Json.Require(m, "title"), "title"),
                Json.AsString(Json.Require(m, "body"), "body")
            );
        }

        public static ReplacePostTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<ReplacePostTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<ReplacePostTyped> FromJSONList(string json)
        {
            var result = new List<ReplacePostTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "ReplacePostTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"id={id}, userId={userId}, title={title}, body={body}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not ReplacePostTyped other) return false;
            return object.Equals(this.id, other.id)
                && object.Equals(this.userId, other.userId)
                && object.Equals(this.title, other.title)
                && object.Equals(this.body, other.body);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.id);
            h.Add(this.userId);
            h.Add(this.title);
            h.Add(this.body);
            return h.ToHashCode();
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/ReplacePostTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class ReplacePostTyped {
    public int id;
    public int userId;
    public String title;
    public String body;

    public ReplacePostTyped(int id, int userId, String title, String body) {
        this.id = id;
        this.userId = userId;
        this.title = title;
        this.body = body;
    }

    public ReplacePostTyped(ReplacePostString s) {
        this.id = Integer.parseInt(s.id);
        this.userId = Integer.parseInt(s.userId);
        this.title = s.title;
        this.body = s.body;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ReplacePostTyped)) return false;
        ReplacePostTyped that = (ReplacePostTyped) o;
        return Objects.equals(id, that.id)
            && Objects.equals(userId, that.userId)
            && Objects.equals(title, that.title)
            && Objects.equals(body, that.body);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, userId, title, body);
    }

    public ReplacePostString toReplacePostString() {
        return new ReplacePostString(String.valueOf(id), String.valueOf(userId), String.valueOf(title), String.valueOf(body));
    }

    public static List<ReplacePostTyped> fromStringList(List<ReplacePostString> list) {
        List<ReplacePostTyped> result = new ArrayList<>();
        for (ReplacePostString s : list) result.add(new ReplacePostTyped(s));
        return result;
    }

    public static List<ReplacePostString> toStringList(List<ReplacePostTyped> list) {
        List<ReplacePostString> result = new ArrayList<>();
        for (ReplacePostTyped t : list) result.add(t.toReplacePostString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("id", id);
        m.put("userId", userId);
        m.put("title", title);
        m.put("body", body);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static ReplacePostTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new ReplacePostTyped(
            Json.asInt(Json.require(m, "id"), "id"),
            Json.asInt(Json.require(m, "userId"), "userId"),
            Json.asString(Json.require(m, "title"), "title"),
            Json.asString(Json.require(m, "body"), "body"));
    }

    public static ReplacePostTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<ReplacePostTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (ReplacePostTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<ReplacePostTyped> fromJSONList(String json) {
        List<ReplacePostTyped> result = new ArrayList<ReplacePostTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "ReplacePostTyped")));
        return result;
    }
}
